File: bdp-master-server/src/main/java/org/bdp/master/server/service/AlertIndexService.java
package org.bdp.master.server.service;

import org.bdp.master.server.domain.AlertIndex;

import java.util.List;

public interface AlertIndexService {

    AlertIndex findOne(Long id);

    AlertIndex findOne(String alertName);

    List<AlertIndex> findAll();

    void save(AlertIndex alertIndex);

    void delete(Long id);

    void loadAll();

}

File: bdp-master-client/src/main/scala/org/bdp/master/client/service/AlertIndexService.scala
package org.bdp.master.client.service

import com.typesafe.scalalogging.LazyLogging
import org.bdp.master.client.domain.AlertIndex
import org.bdp.master.client.util.{JsonDecoder, RedisClient}
import org.bdp.master.client.Constants._

object AlertIndexService extends LazyLogging {

  def getAlertIndexBy(id: Long): AlertIndex = {
    JsonDecoder.decodeAlertIndex(RedisClient.get(s"$ALERT_INDEX_KEYSPACE:$id"))
  }

  def getAlertIndexBy(name: String): AlertIndex = {
    val key = RedisClient.get(s"$INDEX_PREFIX$ALERT_INDEX_KEYSPACE:$name")
    JsonDecoder.decodeAlertIndex(RedisClient.get(key))
  }

}
